### ISSUE TITLE
[BUG] Unable to launch persistent Firefox context

### ISSUE THREAD

**Issue by MaxBex** [microsoft/playwright-dotnet#2070]
**Context:**
- Playwright Version: 1.20.1
- Operating System: Windows
- .NET version: .NET 6
- Browser: Firefox
- Extra: 

**Code Snippet**
```cs
IPlaywright? playwright = await Playwright.CreateAsync();
await using var context = await playwright.Firefox.LaunchPersistentContextAsync(System.IO.Path.Combine(Environment.CurrentDirectory, "appData"), new BrowserTypeLaunchPersistentContextOptions() {
    Headless = false,
    Devtools = false,
    ViewportSize = ViewportSize.NoViewport
}).ConfigureAwait(false);
var page = context.Pages.FirstOrDefault();
await page.GotoAsync("https://playwright.dev/", new PageGotoOptions() { WaitUntil = WaitUntilState.Load });
await page.ScreenshotAsync(new PageScreenshotOptions() { FullPage = true, Path = "Result.png", Type = ScreenshotType.Png });
await context.DisposeAsync();
playwright.Dispose();
```

**Describe the bug**
The browser start but crash after reaching the timeout without navigate.

**Debug info**
```
2022-03-16T16:44:05.459Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'BrowserType',
    initializer: {
      executablePath: 'C:\\Users\\Maximilian\\AppData\\Local\\ms-playwright\\chromium-978106\\chrome-win\\chrome.exe',
      name: 'chromium'
    },
    guid: 'browser-type@e2db15b3318569900f627534376f7089'
  }
}
2022-03-16T16:44:05.461Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'BrowserType',
    initializer: {
      executablePath: 'C:\\Users\\Maximilian\\AppData\\Local\\ms-playwright\\firefox-1319\\firefox\\firefox.exe',
      name: 'firefox'
    },
    guid: 'browser-type@57132167f7c878fd3bc9690076c58299'
  }
}
2022-03-16T16:44:05.462Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'BrowserType',
    initializer: {
      executablePath: 'C:\\Users\\Maximilian\\AppData\\Local\\ms-playwright\\webkit-1616\\Playwright.exe',
      name: 'webkit'
    },
    guid: 'browser-type@d7e8f0b0b2939430847b2eee46186dec'
  }
}
2022-03-16T16:44:05.462Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'Android',
    initializer: {},
    guid: 'android@3d9a0d5075424d4d55990afd07a6c55b'
  }
}
2022-03-16T16:44:05.463Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'Electron',
    initializer: {},
    guid: 'electron@e5df63cb8564085c20770211accc2294'
  }
}
2022-03-16T16:44:05.463Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'LocalUtils',
    initializer: {},
    guid: 'localUtils@979ae22f558bcad60a750d221472c4a9'
  }
}
2022-03-16T16:44:05.464Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'Selectors',
    initializer: {},
    guid: 'selectors@26c1cf06d1a4a0e44712db27fdc5754e'
  }
}
2022-03-16T16:44:05.467Z pw:channel:event {
  guid: '',
  method: '__create__',
  params: {
    type: 'Playwright',
    initializer: {
      chromium: [Object],
      firefox: [Object],
      webkit: [Object],
      android: [Object],
      electron: [Object],
      utils: [Object],
      deviceDescriptors: [Array],
      selectors: [Object],
      preLaunchedBrowser: undefined,
      socksSupport: undefined
    },
    guid: 'Playwright'
  }
}
2022-03-16T16:44:06.148Z pw:browser <launching> C:\Users\Maximilian\AppData\Local\ms-playwright\firefox-1319\firefox\firefox.exe -no-remote -wait-for-browser -foreground -profile C:\Users\Maximilian\source\repos\PlaywrightTest\PlaywrightTest\bin\Debug\net6.0\appData -juggler-pipe about:blank
2022-03-16T16:44:06.155Z pw:browser <launched> pid=24260
2022-03-16T16:44:06.161Z pw:protocol SEND Ôû║ {"id":1,"method":"Browser.enable","params":{"attachToDefaultContext":true}}
2022-03-16T16:44:06.162Z pw:protocol SEND Ôû║ {"id":2,"method":"Browser.getInfo"}
2022-03-16T16:44:06.165Z pw:protocol SEND Ôû║ {"id":3,"method":"Browser.setDownloadOptions","params":{"downloadOptions":{"behavior":"saveToDisk","downloadsDir":"C:\\Users\\Maximilian\\AppData\\Local\\Temp\\playwright-artifacts-zD1Wop"}}}
2022-03-16T16:44:06.165Z pw:protocol SEND Ôû║ {"id":4,"method":"Browser.setColorScheme","params":{"colorScheme":"light"}}
2022-03-16T16:44:06.165Z pw:protocol SEND Ôû║ {"id":5,"method":"Browser.setReducedMotion","params":{"reducedMotion":"no-preference"}}
2022-03-16T16:44:06.165Z pw:protocol SEND Ôû║ {"id":6,"method":"Browser.setForcedColors","params":{"forcedColors":"none"}}
2022-03-16T16:44:06.744Z pw:browser [pid=24260][out]
2022-03-16T16:44:06.744Z pw:browser [pid=24260][out] Juggler listening to the pipe
2022-03-16T16:44:06.811Z pw:protocol ÔùÇ RECV {"id":2,"result":{"version":"Firefox/97.0.1","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:97.0) Gecko/20100101 Firefox/97.0"}}
2022-03-16T16:44:06.812Z pw:protocol ÔùÇ RECV {"id":3}
2022-03-16T16:44:06.812Z pw:protocol ÔùÇ RECV {"id":4}
2022-03-16T16:44:06.812Z pw:protocol ÔùÇ RECV {"id":5}
2022-03-16T16:44:06.812Z pw:protocol ÔùÇ RECV {"id":6}
2022-03-16T16:44:07.627Z pw:protocol ÔùÇ RECV {"id":1}
2022-03-16T16:44:07.911Z pw:protocol ÔùÇ RECV {"method":"Browser.attachedToTarget","params":{"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895","targetInfo":{"targetId":"86b722a3-a9b0-4481-b8e4-51864b43b8e5","type":"page"}}}
2022-03-16T16:44:07.916Z pw:protocol SEND Ôû║ {"method":"Page.setInitScripts","params":{"scripts":[{"script":"","worldName":"__playwright_utility_world__"}]},"id":7,"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895"}
2022-03-16T16:44:08.000Z pw:protocol ÔùÇ RECV {"method":"Page.frameAttached","params":{"frameId":"frame-29"},"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895"}
2022-03-16T16:44:08.002Z pw:protocol ÔùÇ RECV {"method":"Page.navigationCommitted","params":{"frameId":"frame-29","url":"about:blank","name":""},"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895"}
2022-03-16T16:44:08.003Z pw:api   navigated to "about:blank"
2022-03-16T16:44:08.004Z pw:protocol ÔùÇ RECV {"method":"Runtime.executionContextCreated","params":{"executionContextId":"id-1","auxData":{"frameId":"frame-29","name":""}},"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895"}
2022-03-16T16:44:08.005Z pw:protocol ÔùÇ RECV {"method":"Runtime.executionContextCreated","params":{"executionContextId":"id-2","auxData":{"frameId":"frame-29","name":"__playwright_utility_world__"}},"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895"}
2022-03-16T16:44:08.005Z pw:protocol ÔùÇ RECV {"id":7,"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895"}
2022-03-16T16:44:08.244Z pw:protocol ÔùÇ RECV {"method":"Page.navigationStarted","params":{"frameId":"frame-29","navigationId":"17824114278401","url":"about:blank"},"sessionId":"ed64629d-67b6-4984-af16-48cdffd75895"}
2022-03-16T16:44:35.610Z pw:browser [pid=24260] <gracefully close start>
2022-03-16T16:44:35.612Z pw:browser [pid=24260] <kill>
2022-03-16T16:44:35.612Z pw:browser [pid=24260] <will force kill>
2022-03-16T16:44:35.856Z pw:browser [pid=24260] taskkill stderr: OPERAZIONE RIUSCITA: il processo con PID 22968 (processo figlio del

PID 24044) ´┐¢ stato terminato.
OPERAZIONE RIUSCITA: il processo con PID 8300 (processo figlio del

PID 24044) ´┐¢ stato terminato.
OPERAZIONE RIUSCITA: il processo con PID 12180 (processo figlio del

PID 24044) ´┐¢ stato terminato.
OPERAZIONE RIUSCITA: il processo con PID 21692 (processo figlio del

PID 24044) ´┐¢ stato terminato.
OPERAZIONE RIUSCITA: il processo con PID 24044 (processo figlio del

PID 24260) ´┐¢ stato terminato.
OPERAZIONE RIUSCITA: il processo con PID 24260 (processo figlio del

PID 12064) ´┐¢ stato terminato.

2022-03-16T16:44:35.860Z pw:error Error: write EPIPE
    at afterWriteDispatched (node:internal/stream_base_commons:164:15)
    at writeGeneric (node:internal/stream_base_commons:155:3)
    at Socket._writeGeneric (node:net:795:11)
    at Socket._write (node:net:807:8)
    at doWrite (node:internal/streams/writable:408:12)
    at clearBuffer (node:internal/streams/writable:569:7)
    at onwrite (node:internal/streams/writable:461:7)
    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:110:10)
2022-03-16T16:44:35.861Z pw:browser [pid=24260] <process did exit: exitCode=1, signal=null>
2022-03-16T16:44:35.862Z pw:browser [pid=24260] starting temporary directories cleanup
2022-03-16T16:44:35.875Z pw:browser [pid=24260] finished temporary directories cleanup
2022-03-16T16:44:35.875Z pw:browser [pid=24260] <gracefully close end>
Unhandled exception. System.TimeoutException: Timeout 30000ms exceeded.
=========================== logs ===========================
<launching> C:\Users\Maximilian\AppData\Local\ms-playwright\firefox-1319\firefox\firefox.exe -no-remote -wait-for-browser -foreground -profile C:\Users\Maximilian\source\repos\PlaywrightTest\PlaywrightTest\bin\Debug\net6.0\appData -juggler-pipe about:blank
<launched> pid=24260
[pid=24260][out]
[pid=24260][out] Juggler listening to the pipe
============================================================
   at Microsoft.Playwright.Transport.Connection.InnerSendMessageToServerAsync[T](String guid, String method, Object args) in /_/src/Playwright/Transport/Connection.cs:line 161
   at Microsoft.Playwright.Transport.Connection.WrapApiCallAsync[T](Func`1 action, Boolean isInternal)
   at Microsoft.Playwright.Core.BrowserType.LaunchPersistentContextAsync(String userDataDir, BrowserTypeLaunchPersistentContextOptions options) in /_/src/Playwright/Core/BrowserType.cs:line 87
   at Program.<Main>$(String[] args) in C:\Users\Maximilian\source\repos\PlaywrightTest\PlaywrightTest\Program.cs:line 81
   at Program.<Main>(String[] args)
```

**Comment by mxschmitt** [comment]
Could you try if the same reproduces if you are running it with `Headless: true`? If its working headless, then you probably encounter this bug report: https://github.com/microsoft/playwright/issues/12632

**Comment by MaxBex** [comment]
With `Headless = true` it's working. I will follow the other report, thanks.

**Comment by mxschmitt** [comment]
Folding by that in the other issue, it's probably getting fixed in 1.21.
